### PR TITLE
Disallow reading from Delta Lake tables in hive connector

### PR DESCRIPTION
**Pull request by wendigo**

### CODE REVIEW

**Comment by tooptoop4** [comment]
why?

**Comment by findepi** [comment]
> why?

because it works anyway, just in a cryptic way

File: presto-hive/src/main/java/io/prestosql/plugin/hive/HiveErrorCode.java [hunk 1]
```diff
@@ -64,7 +64,8 @@
     HIVE_CORRUPTED_COLUMN_STATISTICS(37, EXTERNAL),
     HIVE_EXCEEDED_SPLIT_BUFFERING_LIMIT(38, USER_ERROR),
     HIVE_UNKNOWN_COLUMN_STATISTIC_TYPE(39, INTERNAL_ERROR),
-    HIVE_TABLE_LOCK_NOT_ACQUIRED(40, EXTERNAL)
+    HIVE_TABLE_LOCK_NOT_ACQUIRED(40, EXTERNAL),
+    HIVE_DELTA_LAKE_TABLE(40, USER_ERROR),
```
Comment (findepi) [inline presto-hive/src/main/java/io/prestosql/plugin/hive/HiveErrorCode.java:68]: just use `HIVE_UNSUPPORTED_FORMAT` instead

File: presto-hive/src/main/java/io/prestosql/plugin/hive/HiveMetadata.java [hunk 2]
```diff
@@ -2296,6 +2303,12 @@ private static void validateBucketColumns(ConnectorTableMetadata tableMetadata)
         }
     }
 
+    private static boolean isDeltaLakeManagedTable(Table table)
```
Comment (findepi) [inline presto-hive/src/main/java/io/prestosql/plugin/hive/HiveMetadata.java:2306]: rm "Managed"

File: presto-hive/src/main/java/io/prestosql/plugin/hive/HiveMetadata.java [hunk 3]
```diff
@@ -341,6 +344,10 @@ public HiveTableHandle getTableHandle(ConnectorSession session, SchemaTableName
             return null;
         }
 
+        if (isDeltaLakeManagedTable(table.get())) {
+            throw new PrestoException(HIVE_DELTA_LAKE_TABLE, "Cannot query Delta Lake managed table");
```
Comment (findepi) [inline presto-hive/src/main/java/io/prestosql/plugin/hive/HiveMetadata.java:348]: rm "managed"

**Comment by electrum** [comment]
@findepi I assume you meant to say that it fails?

**Comment by wendigo** [comment]
> @findepi I assume you meant to say that it fails?

It depends. For partitioned tables it will return 0 results (as partition information is hold in transaction log). For non-partitioned tables it will fail while executing query (in metastore table doesn't have serde defined so it defaults to SEQUENCEFILE).

**Pull request merged**